### PR TITLE
adding v1 events and event subscriptions APIs for webhook capabilities

### PR DESCRIPTION
**Pull request by magaldima**
Associated docs/guides are available here: https://givechariot.readme.io/v1-webhooks/reference/webhooks-and-events

### CODE REVIEW

**Comment by salo1serfati** [comment]
image

This section seems a little complicated. Do services like Stripe/Increase make it easier to verify a signature? Also, I see the signingSecret is optional below. Isn't that needed to be able to secure the webhook above? 
image

**Comment by magaldima** [comment]
@salo1serfati yea the section is complicated but it's basically a word-for-word copy of the documentation from Stripe/Increase. The alternative is for us to create, maintain, and publish an SDK in every language to be able to do automatically upon receiving webhooks which Stripe does but Increase does not do. Regardless, this logic is pretty standard and well documented.

As to the second point about the signingSecret, that's a great observation and point. Platforms that allow you to create event subscriptions allow users to do it from a dashboard where the users can see and grab the signingSecret that is automatically generated (if they don't specify one) to use for verifying their webhooks. We currently don't have a dashboard for FP clients to use so it's purely by API and the external API purposefully does not expose the `signingSecret` in the response of the Event Subscription API because it is private/sensitive (one of those things you should see once upon creation). I'm happy to make the `signingSecret` property required on the Create route

**Comment by magaldima** [comment]
@salo1serfati I've updated the description the property and the guide to make it uber clear that you need to specify that field in order to validate webhooks which while not required is highly recommended

File: specs/v1.yaml [hunk 1]
```diff
@@ -907,6 +1146,84 @@ components:
...
+          description: The date and time the event was created
+          example: "2024-01-19T18:48:56.37Z"
+          readOnly: true
+        associated_object_id:
+          type: string
+          description: The unique identifier for the associated object
+          example: "4d06d393-1f14-46cf-b02d-8db17d7ed06a"
+          readOnly: true
+        associated_object_type:
+          type: string
```
[the remark this answers is not in the document]
Comment (magaldima) [reply specs/v1.yaml:1182]: for now `string` is okay but we are recording this in our API docs guide

**Review by Neilshweky — APPROVED** [review]

**Pull request merged**